unit-tests/func/hw-errors/error-common.h

```diff
@@ -0,0 +1,47 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#include <concurrency.h>
+
+using namespace rs2;
+
+// This function loop over all posiable hw errors use hw-monitor command to trigger them
+// and check that we call to user callback with the error
+void validate_errors_handling( const rs2::device & dev,
+                               const std::map< uint8_t, std::pair< std::string, rs2_log_severity > > & error_report )
+{
+    auto depth_sens = dev.first< rs2::depth_sensor >();
+
+    REQUIRE( depth_sens.supports( RS2_OPTION_ERROR_POLLING_ENABLED ) );
+
+    std::string notification_description;
+    rs2_log_severity severity;
+    std::condition_variable cv;
+    std::mutex m;
+
+
+    depth_sens.set_notifications_callback( [&]( rs2::notification n ) {
+        std::lock_guard< std::mutex > lock( m );
+
+        notification_description = n.get_description();
+        severity = n.get_severity();
+        cv.notify_one();
+    } );
+
+    REQUIRE_NOTHROW( depth_sens.set_option( RS2_OPTION_ERROR_POLLING_ENABLED, 1 ) );
+
+    for( auto i = error_report.begin(); i != error_report.end(); i++ )
+    {
+        trigger_error_or_exit( dev, i->first );
```

Comment:
How does the generic error-common.h know about trigger_error_or_exit()?
Is inclusion of l500-error-common.h mandatory before this one?

Reply:
fixed